DDD-9.md

```diff
@@ -0,0 +1,111 @@
+
+Lastly, one of the more technical critiques is everyone believes the Oracle connector's code is significantly complex.
+If we eliminate the need to buffer transactions entirely, that significantly reduces the complexity and integration with cache providers, simplifies the configuration, and slims the documentation so that perhaps it's easier to digest with less moving parts. 
+
+## Goals
+
+* Reduce code complexity, eliminating the need for on-heap and off-heap cache integrations.
+* Reduce connector memory footprint
+* Avoid issues with split transactions that yield _unsupported_ operations.
+* Provides an easy way to manage transaction-level sequences for restarts.
+* Remove the need for `rac.nodes`.
+* Remove the need for a flush table and flush strategies.
+
+## Additional benefits
+
+* Data set would be provided in committed, chronological order.
+* Implementation can use simple read, parse, dispatch pattern similar to other connectors.
+* Read strategy aligns closely with the current one used when `lob.enabled` is set to `true`.
+
+## Requirements
+
+* LogMiner session must be started to return only committed data.
+* The LogMiner resume SCN is used to control the low watermark in the offsets.
+* Transactions committed after the resume SCN must have its id and SCN maintained in the offsets.
+* LogMiner query should exclude any transaction already committed since the resume SCN.
+
+## Proposed changes
+
+### Offsets
+
+Currently, the Oracle connector maintains two critical values in the offsets, `scn` and `commit_scn`, shown here:
+
+```json
+{
+    "scn": "123456",
+    "commit_scn": "123456:1:txId1-txId2,123457:2:txId4"
+}
+```
+
+Moving forward, the connector needs to maintain the resume SCN and each committed transaction id since the resume SCN and its commit SCN.
+In addition, need to maintain the last processed transaction id and event sequence.
+I would also prefer to move away from _encoded_ values to the more natural nested `Struct`s style.
+
+I'm not sure how hard this makes editing the offsets for users, but this what I have in mind:
+
+```json
+{
+  "resume_scn": "123456",
+  "txId": "123456", // current transaction being processed
+  "txSeq": 25, // last event in current transaction received
+  "committed_transactions": [{
+    "id": "a748391bc3123",
+    "thread": 1,
+    "scn": "123478"
+  }]
+}
+```
+
+### LogMiner query
+
+The LogMiner query must be adjusted to include a new predicate condition to exclude transactions in `committed_transactions`.
+The query change would include something like the following:
+
+```sql
+AND UPPER(RAWTOHEX(XID)) NOT IN ( <committed-transaction.id(s) in upper-case> )
+```
+
+The construction of the in-clause should use the same logic for the `table.include.list` to wrap the predicate in a disjunction is the list of values exceeds 1000 values, since Oracle does not permit creating in-clauses with more than 1000 elements.
```

Comment:
```suggestion
The construction of the in-clause should use the same logic for the `table.include.list` to wrap the predicate in a disjunction if the list of values exceeds 1000 values, since Oracle does not permit creating in-clauses with more than 1000 elements.
```